lib/bolt/cli.rb

```diff
@@ -654,8 +654,21 @@ def generate_types
     # Initializes a specified directory as a Bolt project and installs any modules
     # specified by the user, along with their dependencies
     def initialize_project
-      project    = Pathname.new(File.expand_path(options[:object] || Dir.pwd))
-      config     = project + 'bolt.yaml'
+      name       = options[:object] || File.basename(Dir.pwd)
+      name_regex = /^[a-z][a-z0-9_]*$/
```

Comment:
```suggestion
      name_regex = /\A[a-z][a-z0-9_]*\Z/
```

Reply:
Also, since this regex is also used in `Bolt::Project`, I think it might make sense to define it as a constant so it can be reused elsewhere. I made this change in #2004 for reference: https://github.com/puppetlabs/bolt/pull/2061/files#diff-5fb12f608b24f29bc6e72995332d2c24R11

Reply:
Is it possible to open a separate PR for it if it's going to be used here too? I don't want to duplicate work, or for this to unnecessarily rely on #2004.

Reply:
Ya, I'll throw something up real quick.

Reply:
https://github.com/puppetlabs/bolt/pull/2063 has the change

Reply:
Ugh, I just realized we already have this in https://github.com/puppetlabs/bolt/blob/8d3e2f207daf162aeceec12621553ecdf9f5d48a/lib/bolt/module.rb#L6. I'm just going to modify this PR to use that instead.

Reply:
And in a way that seems better, since the only reason we limit the project name is because it needs to be loaded as a module.